### DISCUSSION TITLE
When use PrismaClient select or other option to filter result the code highlighted with error.

### DISCUSSION THREAD

**Discussion by Historaj** [prisma/prisma#12711] [Q&A] [answered]
When using PrismaClient select or other option to filter the result the code is highlighted with an error. But expected functionality working fine. 

Environment:
VScode1.66.0
Prisma: 3.11.1
OS: Windows10

![Snipping-tool](https://user-images.githubusercontent.com/97240227/161913700-b1b3acb3-dc89-4d13-82d8-da79ea6bb2c9.PNG)

**Comment by janpio** [comment]
This is probably from the TypeScript support in your editor, not the Prisma VS Code extension.

Anyway:
What is the error shown when hovering that red squiggly underlined text?

**Comment by Historaj** [comment]
I attached the error screenshot
![image](https://user-images.githubusercontent.com/97240227/161980604-baedfc8c-6a10-4155-aa04-bb7253594a3f.png)

**Comment by Jolg42** [comment, marked as answer]
It doesn't look related to the Prisma extension but to TypeScript language server here, you should get the same error when calling `tsc`.

If you remove the typed return `Promise<Client[]>` it should work again.

Because it looks like you typed the return to `Promise<Client[]>` and it’s apparently the full `Client` "model", while the Prisma query only select some fields (ClientCode and ClientName)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
